keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -20,11 +20,32 @@ package com.netflix.spinnaker.keel.api
 /**
  * An object which is located in an account and region
  */
-interface Locatable<T : RegionSpec> : ResourceSpec {
-  val locations: Locations<T>
+interface Locatable<T : Locations<*>> : ResourceSpec {
+  val locations: T
 }
 
-data class Locations<T : RegionSpec>(
-  val accountName: String,
+interface Locations<T : RegionSpec> {
+  val accountName: String
+  /**
+   * If not specified here, this should be derived from the [SubnetAwareLocations.subnet] (if
+   * present) or use a default VPC name.
+   */
+  val vpcName: String?
   val regions: Set<T>
-)
+}
+
+data class SubnetAwareLocations(
+  override val accountName: String,
+  override val vpcName: String?,
+  /**
+   * If not specified here, this should be derived from a default subnet purpose using [vpcName].
+   */
+  val subnet: String?,
+  override val regions: Set<SubnetAwareRegionSpec>
+) : Locations<SubnetAwareRegionSpec>
+
+data class SimpleLocations(
+  override val accountName: String,
+  override val vpcName: String?,
+  override val regions: Set<SimpleRegionSpec>
+) : Locations<SimpleRegionSpec>
```

Comment:
Instead of one type of `Locations` and 2 types of `RegionSpec` there are now 2 types of `Locations`. This leads to more consistent YAML between resource types that specify zones and those that do not.